test/integration/node-specific/bson-options/utf8_validation.test.ts

```diff
@@ -153,3 +65,182 @@ describe('class MongoDBResponse', () => {
     }
   );
 });
+
+describe('utf8 validation with cursors', function () {
```

Comment:
Just want to make the fact that we are explicitly testing invalid documents in all the below cases more visible
```suggestion
describe('parsing of utf8-invalid documents with cursors', function () {
```